pypose/lietensor/operation.py

```diff
@@ -471,14 +472,20 @@ def backward(ctx, grad_output):
 class SO3_Act(torch.autograd.Function):
 
     @staticmethod
-    def forward(ctx, X, p):
+    def forward(X, p):
         Xv, Xw = X[..., :3], X[..., 3:]
         uv = torch.linalg.cross(Xv, p, dim=-1)
         uv += uv
         out = p + Xw * uv + torch.linalg.cross(Xv, uv, dim=-1)
-        ctx.save_for_backward(X, out)
         return out
 
+    @staticmethod
+    def setup_context(ctx, inputs, output):
+        X, p = inputs
+        out = output
+        ctx.save_for_backward(X, out)
+        return
+
```

Comment:
Seems like this is a nice bit of cleanup that is tangent to the context manager we are adding. I am not very familiar with this part of the code base, but would adding a link to PyTorch's documentation about [extending torch.autograd](https://pytorch.org/docs/stable/notes/extending.html) be helpful here? It took me a while to understand why we can safely remove the `ctx` from `forward` by adding the `setup_context` here, and this documentation helped me personally.

Reply:
It's just a new requirement from torch 2.0